addons/survey/static/src/js/survey_result.js

```diff
@@ -455,19 +463,25 @@ publicWidget.registry.SurveyResultWidget = publicWidget.Widget.extend({
     // Handlers
     // -------------------------------------------------------------------------
 
-    /**
+     /**
+     * Add an answer filter by updating the URL and redirecting.
      * @private
      * @param {Event} ev
      */
     _onSurveyAnswerFilterClick: function (ev) {
-        var cell = $(ev.target);
-        var row_id = cell.data('row_id') | 0;
-        var answer_id = cell.data('answer_id');
-
-        var params = new URLSearchParams(window.location.search);
-        var filters = params.get('filters') ? params.get('filters') + "|" + row_id + ',' + answer_id : row_id + ',' + answer_id;
-        params.set('filters', filters);
+        let params = new URLSearchParams(window.location.search);
+        params = this._updateURLParamsAnswerFilters(params, 'add', this._getQuestionEventFilter(ev));
+        window.location.href = window.location.pathname + '?' + params.toString();
+    },
 
+    /**
+     * Remove an answer filter by updating the URL and redirecting.
+     * @private
+     * @param {Event} ev
+     */
+    _onClearAnswerFilterClick: function (ev) {
+        let params = new URLSearchParams(window.location.search);
+        params = this._updateURLParamsAnswerFilters(params, 'remove', this._getQuestionEventFilter(ev));
```

Comment:
I would rather do:
`params.set('filter', this._prepareFilterParameters(params.get('filter'), 'remove', questionFilter)`

It's usually always a bit more clear when the method has a single "job" and when it's passed only what it needs.